src/analytics/events.ts

```diff
@@ -65,6 +68,21 @@ export async function fetchAllCliEvents(): Promise<PosthogEvent[]> {
   events = [...newEvents, ...events];
   await saveCachedEvents(events);
 
+  // NOTE: Sometimes, likely due to rate limiting from PostHog side, `isThereMore` will falsely be
+  //   set to `false` even when there is more data. To handle that, we check here if we actually got
+  //   all the events, by checking if the oldest event we fetched is indeed old enough.
+  const oldestFetchedEventTimestamp = getOldestEventTimestampOrNull(events);
+  const didWeFetchAllOldEvents =
+    oldestFetchedEventTimestamp &&
+    moment(oldestFetchedEventTimestamp).isSameOrBefore(
+      moment(OLDEST_EVENT_TIMESTAMP),
+    );
```

Comment:
```suggestion
    !!oldestFetchedEventTimestamp &&
    moment(oldestFetchedEventTimestamp).isSameOrBefore(
      moment(OLDEST_EVENT_TIMESTAMP),
    );
```

To convert `null` to `false`

Reply:
I can do this, but won't `&&` already convert `oldestFetchedEventTimestamp` to `bool`, using the same rules as you would get by doing `!!`?
From what I know, the point of `!!` is that we force casting it to bool (and during that JS does the conversion with its crazy rules), but since this is argument of `&&`, this is happening already due to that.

Reply:
Got it nice, thanks!